### PR TITLE
Docs and test code for  `PrioritizedEpisodeReplayBuffer`.

### PR DESCRIPTION
**Pull request by simonsays1980**
## Why are these changes needed?

We move from plain timesteps to episodes in the buffer. This needs some rigorous documentation with appropriate test code to show users how to use a class. 

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: rllib/utils/replay_buffers/prioritized_episode_replay_buffer.py [hunk 1]
```diff
@@ -16,6 +16,104 @@
 
 
 class PrioritizedEpisodeReplayBuffer(EpisodeReplayBuffer):
+    """Prioritized Replay Buffer that stores episodes by their ID.
+
+    This replay buffer stores episode data (more specifically `EpisodeV2`
```
Comment (sven1977) [inline rllib/utils/replay_buffers/prioritized_episode_replay_buffer.py:21]: nit: `SingleAgentEpisode` objects.

EpisodeV2 is wrong here. It's an old-API-stack class.

File: rllib/utils/replay_buffers/prioritized_episode_replay_buffer.py [hunk 2]
```diff
@@ -16,6 +16,104 @@
...
+    added at a later time, the buffer will automatically concatenate the new fragment to
+    the original episode. This way, episodes can be completed via subsequent `add`
+    calls.
+
+    Sampling returns batches of size B (number of 'rows'), where each row is a tuple
+    of the form
+
+    `(o_t, a_t, sum(r_t+1:t+n), o_t+n)`
+
+    where `o_t` is the observation in `t`, `a_t` the action chosen at observation `o_t`,
```
Comment (sven1977) [inline rllib/utils/replay_buffers/prioritized_episode_replay_buffer.py:40]: nice! this is very clear now.

File: rllib/utils/replay_buffers/prioritized_episode_replay_buffer.py [hunk 3]
```diff
@@ -16,6 +16,104 @@
...
+    `o_t+n` is the observation `n` timesteps later and `sum(r_t+1:t+n)` is the sum of
+    all rewards collected over the time steps between `t+1` and `t+n`. The `n`-step can
+    be chosen freely when sampling and defaults to `1`. If `n_step` is a tuple it is
+    sampled uniformly across the interval defined by the tuple (for each row in the
+    batch).
+
+    Each batch contains in addition to the data tuples presented above two further
```
Comment (sven1977) [inline rllib/utils/replay_buffers/prioritized_episode_replay_buffer.py:47]: nit: "Each batch contains - in addition to the data tuples presented above - two ..."

File: rllib/utils/replay_buffers/prioritized_episode_replay_buffer.py [hunk 4]
```diff
@@ -24,6 +122,17 @@ def __init__(
         batch_length_T: int = 1,
         alpha: float = 1.0,
     ):
+        """Initializes a `PrioritizedEpisodeReplayBuffer` object
+
+        Args:
+            capacity: The total number of timesteps to be storable in this buffer.
+                Will start ejecting old episodes once this limit is reached.
+            batch_size_B: The number of rows in a SampleBatch returned from `sample()`.
+            batch_length_T: The length of each row in a SampleBatch returned from
```
Comment (sven1977) [inline rllib/utils/replay_buffers/prioritized_episode_replay_buffer.py:131]: Can we specify here, how this arg is related to n-step? My understanding is: Not at all, correct?

So, batch_length_T is the same as in DreamerV3, meaning if >1, then my returned batch will contain B rows, each row consisting of T consecutive-in-an-episode(!) n-step tuples, correct?

If so, can we give these details here in the description of the arg?

File: rllib/utils/replay_buffers/prioritized_episode_replay_buffer.py [hunk 5]
```diff
@@ -259,8 +371,8 @@ def sample(
         next_observations = [[] for _ in range(batch_size_B)]
         actions = [[] for _ in range(batch_size_B)]
         rewards = [[] for _ in range(batch_size_B)]
-        is_terminated = [[False] * batch_length_T for _ in range(batch_size_B)]
-        is_truncated = [[False] * batch_length_T for _ in range(batch_size_B)]
+        is_terminated = [[False] for _ in range(batch_size_B)]
```
Comment (sven1977) [inline rllib/utils/replay_buffers/prioritized_episode_replay_buffer.py:374]: Can you explain, why this fix? B/c a returned batch only has one terminated/truncated flag per batch row?

**Review by sven1977 — APPROVED** [review]
LGTM, just a few questions and nits to fix before we can merge.

File: rllib/utils/replay_buffers/prioritized_episode_replay_buffer.py (hunk 4 above)
Comment (simonsays1980) [reply rllib/utils/replay_buffers/prioritized_episode_replay_buffer.py:131]: Ar rhis moment `batch_öength_T=1` is the only possible option. As this is a method inherited from `EpisodeReplayBuffer` this argument has to be there.

File: rllib/utils/replay_buffers/prioritized_episode_replay_buffer.py (hunk 5 above)
Comment (simonsays1980) [reply rllib/utils/replay_buffers/prioritized_episode_replay_buffer.py:374]: Mainly because each batch row has only a single tuple. See my comment above.

**Pull request merged**